pgmpy/tests/test_models/test_MarkovChain.py

```diff
@@ -107,6 +107,25 @@ def test_add_variable_new(self):
         self.assertEqual(model.cardinalities['p'], 3)
         self.assertDictEqual(model.transition_models['p'], {})
 
+    def test_copy(self):
+        model = MC(self.variables, self.card, self.start_state)
+        copy = model.copy()
+        self.assertIsInstance(copy, MC)
+        self.assertIsNot(copy, model)
+        self.assertListEqual(hf.recursive_sorted(model.variables), hf.recursive_sorted(copy.variables))
+        self.assertDictEqual(model.cardinalities, copy.cardinalities)
+        self.assertDictEqual(model.transition_models, copy.transition_models)
+        self.assertListEqual(model.state, copy.state)
+        model = MC(['a'], [2])
+        self.assertTrue(len(model.variables)<len(copy.variables))
+        self.assertListEqual(hf.recursive_sorted(self.variables),hf.recursive_sorted(copy.variables))
+        self.assertTrue(len(model.cardinalities)<len(copy.cardinalities))
+        self.assertDictEqual(self.cardinalities, self.cardinalities)
+        self.assertTrue(len(model.transition_models)<len(copy.cardinalities))
+        self.assertDictEqual(self.cardinalities, self.cardinalities)
+        self.assertNotEqual(model.state, copy.state)
+        self.assertListEqual(self.start_state, copy.state)
+
```

Comment:
@anchitjain1234 We don't need test like this. Something like, if I change any value in copy.variables, it doesn't change things in the original. You are creating all together a new instance of MarkovChain.